FstCodec exception when copying session map in changeSessionId

We’re currently switching from Lettuce to Redisson to handle our Spring Sessions. We’re using `FstCodec`, because `JacksonJsonCodec` doesn’t support all of the objects we need to serialize.

We’re getting an exception processing the [response from Redis when reading the old map](https://github.com/redisson/redisson/blob/6cd9b28d9cf17a5d951b28b096634f2d7171fa3b/redisson/src/main/java/org/redisson/spring/session/RedissonSessionRepository.java#L208) (stack trace at end).

It looks like the issue is that the map’s codec isn’t being passed to the batch command, so `FstCodec` is being used for both keys and values, rather than the [map’s `CompositeCodec(StringCodec, FstCodec)`](https://github.com/redisson/redisson/blob/f88365f22969eb4329981d72caff7b6ef7d20330/redisson/src/main/java/org/redisson/spring/session/RedissonSessionRepository.java#L71).

```
org.redisson.client.RedisException: Unexpected exception while processing command
        at org.redisson.command.CommandAsyncService.convertException(CommandAsyncService.java:353)
        at org.redisson.command.CommandAsyncService.get(CommandAsyncService.java:150)
        at org.redisson.command.CommandBatchService.execute(CommandBatchService.java:160)
        at org.redisson.RedissonBatch.execute(RedissonBatch.java:209)
        at org.redisson.spring.session.RedissonSessionRepository$RedissonSession.changeSessionId(RedissonSessionRepository.java:211)
        at org.springframework.session.web.http.SessionRepositoryFilter$SessionRepositoryRequestWrapper.changeSessionId(SessionRepositoryFilter.java:274)
        at javax.servlet.http.HttpServletRequestWrapper.changeSessionId(HttpServletRequestWrapper.java:262)
…
Caused by: java.io.IOException: java.lang.NullPointerException
        at org.nustaq.serialization.FSTObjectInput.readObject(FSTObjectInput.java:247)
        at org.redisson.codec.FstCodec$1.decode(FstCodec.java:250)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:368)
        at org.redisson.client.handler.CommandDecoder.decodeList(CommandDecoder.java:412)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:377)
…
Caused by: java.lang.NullPointerException
        at org.nustaq.serialization.FSTObjectInput.readObjectWithHeader(FSTObjectInput.java:357)
        at org.nustaq.serialization.FSTObjectInput.readObjectInternal(FSTObjectInput.java:331)
        at org.nustaq.serialization.FSTObjectInput.readObject(FSTObjectInput.java:311)
        at org.nustaq.serialization.FSTObjectInput.readObject(FSTObjectInput.java:245)
        at org.redisson.codec.FstCodec$1.decode(FstCodec.java:250)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:368)
        at org.redisson.client.handler.CommandDecoder.decodeList(CommandDecoder.java:412)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:377)
        at org.redisson.client.handler.CommandDecoder.decodeCommandBatch(CommandDecoder.java:266)
        at org.redisson.client.handler.CommandDecoder.decodeCommand(CommandDecoder.java:207)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:134)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:104)
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:498)
        at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:366)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
```

--- a/redisson/src/main/java/org/redisson/spring/session/RedissonSessionRepository.java
+++ b/redisson/src/main/java/org/redisson/spring/session/RedissonSessionRepository.java
@@ -205,7 +205,7 @@ public class RedissonSessionRepository implements FindByIndexNameSessionReposito
             RBatch batch = redisson.createBatch(BatchOptions.defaults());
             batch.getBucket(getExpiredKey(oldId)).remainTimeToLiveAsync();
             batch.getBucket(getExpiredKey(oldId)).deleteAsync();
-            batch.getMap(map.getName()).readAllMapAsync();
+            batch.getMap(map.getName(), map.getCodec()).readAllMapAsync();
             batch.getMap(map.getName()).deleteAsync();
 
             BatchResult<?> res = batch.execute();

